{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "setting.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "mount_file_id": "1agRc3igrr7pEcT0ffgVTwqWq8lXe1qYj",
      "authorship_tag": "ABX9TyMSrbcqzI3CgQ7plWvg6ZfL",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/heiyiyayoo/shujuji/blob/main/setting.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "sXcOvyVLBw55"
      },
      "source": [
        "挂载云盘"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qjvInuGgEVeq",
        "outputId": "98bc05e2-21a6-4e44-d158-b0fb104d2dce"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount(\"/content/drive\")"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GKIg_t2MEd--",
        "outputId": "f8621bd6-c041-40a1-9275-1733de17cdc4"
      },
      "source": [
        "# 切换路径\n",
        "#%cd \"drive/MyDrive/XXX\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z_g0Se7HB0Bg",
        "outputId": "5b1ce06c-121f-4bad-f426-9581888c37b1"
      },
      "source": [
        "!nvidia-smi"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "RGcpLNuQT-R5"
      },
      "source": [
        "pip install d2l torch torchvision"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "CxvOeCU5T_sT"
      },
      "source": [
        "# 私钥公钥GitHub  # 私钥\n",
        "key =  '''-----BEGIN OPENSSH PRIVATE KEY-----\n",
        "b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAABlwAAAAdzc2gtcn\n",
        "NhAAAAAwEAAQAAAYEAyOYuYtkX72AopxKYD2RJt2MNHjtLjvypFvg+KsTVpgZzGmsskfoG\n",
        "sof63Myvg2vbu014rQfKiSlMlg76NnO4QqWp8pvtS20LcWX6vZHGck1zEEBLguqwoFhlIq\n",
        "oMWbrzBbPcq8xbukKWJJtptyX15xfPCGU8sMxCukkS3MURxA+UbtB+8AE7SNLeKoCDyRek\n",
        "LbnXARYoM37NzUalq851JpjDiKS0DV+ZAyOp/VWnr2PLKgZQD2K0UEENlu6F8ctkp68vpS\n",
        "ZQSxuCKWpC0rm4bdaP0Gn9MImVCWhYsMKkTyWhBipkvcUQDAJoMl2tqWEsH8wb56Kskaa1\n",
        "mHNS+lC7bDxjaC2xAVGTDhOx7ILAnin4tZ4/8jOnt2jDdzuDJ2hGLXUXKG34h3Vvt0FlbV\n",
        "wDZ5nqi14TnGWVAkp66Bjd4ONLvbztA+N3FmG0KRgGMzA88+Kvh2y7eWz3uy1mxUoAsuSd\n",
        "2GiIe135gDmOtLHNYak0ecAoXyukf1yfZr0H03KhAAAFkFKYafBSmGnwAAAAB3NzaC1yc2\n",
        "EAAAGBAMjmLmLZF+9gKKcSmA9kSbdjDR47S478qRb4PirE1aYGcxprLJH6BrKH+tzMr4Nr\n",
        "27tNeK0HyokpTJYO+jZzuEKlqfKb7UttC3Fl+r2RxnJNcxBAS4LqsKBYZSKqDFm68wWz3K\n",
        "vMW7pCliSbabcl9ecXzwhlPLDMQrpJEtzFEcQPlG7QfvABO0jS3iqAg8kXpC251wEWKDN+\n",
        "zc1GpavOdSaYw4iktA1fmQMjqf1Vp69jyyoGUA9itFBBDZbuhfHLZKevL6UmUEsbgilqQt\n",
        "K5uG3Wj9Bp/TCJlQloWLDCpE8loQYqZL3FEAwCaDJdralhLB/MG+eirJGmtZhzUvpQu2w8\n",
        "Y2gtsQFRkw4TseyCwJ4p+LWeP/Izp7dow3c7gydoRi11Fyht+Id1b7dBZW1cA2eZ6oteE5\n",
        "xllQJKeugY3eDjS7287QPjdxZhtCkYBjMwPPPir4dsu3ls97stZsVKALLkndhoiHtd+YA5\n",
        "jrSxzWGpNHnAKF8rpH9cn2a9B9NyoQAAAAMBAAEAAAGAKbO+RT6TPVlUlvnBHtDOjIa/Hl\n",
        "ton9HKt73oCbC6OLB2DDHbGqIv6bdYV6UVec5V2leGiURowngUMcICJnN2+D6zPG0poNKN\n",
        "4YgtuFQeQXPaQrFV+2MELwSqWbu6vJWWBMUFNZRCJjPUPt0T4LS8iq/ihfed1YVRBKJOjA\n",
        "1MVsp1IVuePevDhqZe/6uVr5R8vBErlEopJidYSRg1I6jDoLNtQ1wNFAmYFZK9opy5Jiss\n",
        "7RhKXic7Dr2yOHZiz4JMj7q5fZ9SHqigWeWJ1MTj8jwRfU0JXRCDS2w2EmrxpNgNjVxdDI\n",
        "6LP4aBJqZLvPwn7MhdDrzQHQxJtJPvJmgRijBHuoe8n4PEAJynFeQqJa+7lPlfCB11ZrN/\n",
        "hMUA54v3jnl7XVK/rx8TRotNe6jEvlsc8MUmXcgGn/JEH2T9PH39TLrKxFcbcePz5gENkH\n",
        "JsJQk0Fa5MhYXn2Bn9AQwPo3H3R/SliI/sRqKRMTnrzUI3B3FIwNYWW+ySilwhm2ABAAAA\n",
        "wFRnvQBTQ23P79iqOX4BxwdA26NiqhCEpk7GgY08+X4l7L6e8Sb7Jsr3xX/9M/UUDCg59V\n",
        "bBzVhjRgYxF+4+ZaMT0x9sGywu3v4NKcUNqurEIr0oV477kTXtfqFq2naFiNRo8VW0i4ez\n",
        "NuSBEDlPNvJtcPUkD0bJ+nMKTLxmbtFlHqrEApdVNCpS6TCPQpOH7xpb2ORtuD1OdxF9cn\n",
        "7ZWSQ1H4P/2TaiuDZUpTUZfzgRfI0wrR0E5C2N23Qmx88CHAAAAMEA//hUbps1YIE3Jal0\n",
        "l/WQ/QfaHksTxrz0t/YJQT/+QKB135DpEWkapYzuzH/Vbt2ekNjtu3lmiTRBxkJZZgvvJX\n",
        "H0+HNWgRX7ww7oySJiWYLJH8zs07pNMzcaz5562D7vtH6/5U3JFdJFg5jmD5coMqiSimXN\n",
        "882ZUHkwW4fWutkmeKMqz0UPxC9/zaLPRxxnQ/x4MzIlDmjx+ph5QEhDDKB5kQgzmwBnKj\n",
        "YebYsP+r520QMoDvkfwRPqluINUFxhAAAAwQDI7DOAJOQBgkzGEeDtyXwjWjGkmXrMmJtG\n",
        "sSJfm90Vo5vOa4WXhDuXDTW06Z4OjpnPYJdYKP59Vj+3cFvqZQPvcOWdRw2DArHIG7/U9Q\n",
        "eYZTlIMb97k4UKw2s6TwTpYJqywGlZl599Gh/vL/yQKKjKgshP0MVDwm3OtIk4Kmub2tdP\n",
        "8CSG7lizFR2p35T74oU7RxHwmWOJmxFOb1imVQ3SzAgXR2N02YdfmEEpXD0nbLrrDAh6Xx\n",
        "g9D+ej/y9cvkEAAAAUMTQ1NjkwNjM3MUBlbWFpbC5jb20BAgMEBQYH\n",
        "-----END OPENSSH PRIVATE KEY-----\n",
        "'''\n",
        "! mkdir -p /content/drive/MyDrive/root/.ssh\n",
        "with open(r'/content/drive/MyDrive/root/.ssh/id_rsa', 'w', encoding='utf8') as fh:\n",
        "    fh.write(key)\n",
        "! chmod 600 /content/drive/MyDrive/root/.ssh/id_rsa\n",
        "! ssh-keyscan github.com >> /content/drive/MyDrive/root/.ssh/known_hosts  # 这一步是将 gitlab.com 的指纹写到 .ssh/known_hosts 里，避免第一次运行时的yes/no交互"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "W9a3h9VUAeIi"
      },
      "source": [
        "!git config --global user.email \"1456906371@qq.com\"\n",
        "!git config --global user.name \"heiyiyayoo\""
      ],
      "execution_count": 35,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JBnNt-KpAPTR",
        "outputId": "e35a541b-0940-4840-f216-f71f01ef9fc8"
      },
      "source": [
        "!ssh -T git@github.com"
      ],
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory\n",
            "Could not create directory '/root/.ssh'.\r\n",
            "Host key verification failed.\r\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 880
        },
        "id": "9zpG7GFfAWIg",
        "outputId": "65341056-f070-4001-a5d0-d42a43fbbc7f"
      },
      "source": [
        "pwd"
      ],
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "ERROR:root:Internal Python error in the inspect module.\n",
            "Below is the traceback from this internal error.\n",
            "\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py\", line 2882, in run_code\n",
            "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
            "  File \"<ipython-input-41-f70ce4f56b66>\", line 1, in <module>\n",
            "    get_ipython().magic('pwd ')\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py\", line 2160, in magic\n",
            "    return self.run_line_magic(magic_name, magic_arg_s)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py\", line 2081, in run_line_magic\n",
            "    result = fn(*args,**kwargs)\n",
            "  File \"<decorator-gen-83>\", line 2, in pwd\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/magic.py\", line 188, in <lambda>\n",
            "    call = lambda f, *a, **k: f(*a, **k)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/magics/osm.py\", line 244, in pwd\n",
            "    return py3compat.getcwd()\n",
            "FileNotFoundError: [Errno 2] No such file or directory\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py\", line 1823, in showtraceback\n",
            "    stb = value._render_traceback_()\n",
            "AttributeError: 'FileNotFoundError' object has no attribute '_render_traceback_'\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/ultratb.py\", line 1132, in get_records\n",
            "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/ultratb.py\", line 313, in wrapped\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/usr/local/lib/python3.7/dist-packages/IPython/core/ultratb.py\", line 358, in _fixed_getinnerframes\n",
            "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
            "  File \"/usr/lib/python3.7/inspect.py\", line 1502, in getinnerframes\n",
            "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
            "  File \"/usr/lib/python3.7/inspect.py\", line 1460, in getframeinfo\n",
            "    filename = getsourcefile(frame) or getfile(frame)\n",
            "  File \"/usr/lib/python3.7/inspect.py\", line 696, in getsourcefile\n",
            "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
            "  File \"/usr/lib/python3.7/inspect.py\", line 725, in getmodule\n",
            "    file = getabsfile(object, _filename)\n",
            "  File \"/usr/lib/python3.7/inspect.py\", line 709, in getabsfile\n",
            "    return os.path.normcase(os.path.abspath(_filename))\n",
            "  File \"/usr/lib/python3.7/posixpath.py\", line 383, in abspath\n",
            "    cwd = os.getcwd()\n",
            "FileNotFoundError: [Errno 2] No such file or directory\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "error",
          "ename": "FileNotFoundError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3xPzsiKlFvv0",
        "outputId": "3dc10e47-c49a-41c7-b9f0-9817d8629077"
      },
      "source": [
        "cd /root/.ssh"
      ],
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/root/.ssh\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "B8pPvXqmGBZK",
        "outputId": "71a52a11-269b-4393-b697-c9af840ed589"
      },
      "source": [
        "!ls"
      ],
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bZsHch2THS60"
      },
      "source": [
        "!rm -r /root"
      ],
      "execution_count": 39,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "xtMGIY0iHUlE"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}